File: tests/src/test/java/org/veri/be/support/assertion/ExceptionAssertions.java
package org.veri.be.support.assertion;

import org.junit.jupiter.api.function.Executable;
import org.veri.be.lib.exception.ApplicationException;
import org.veri.be.lib.exception.ErrorCode;

import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
import static org.junit.jupiter.api.Assertions.assertThrows;

public final class ExceptionAssertions {

    private ExceptionAssertions() {
    }

    public static ApplicationException assertApplicationException(
            Executable executable,
            ErrorCode expected
    ) {
        ApplicationException exception = assertThrows(ApplicationException.class, executable);
        assertThat(exception.getErrorCode()).isEqualTo(expected);
        return exception;
    }
}

File: tests/src/test/kotlin/org/veri/be/support/assertion/ExceptionAssertions.kt
package org.veri.be.support.assertion

import org.assertj.core.api.AssertionsForClassTypes.assertThat
import org.junit.jupiter.api.Assertions.assertThrows
import org.junit.jupiter.api.function.Executable
import org.veri.be.lib.exception.ApplicationException
import org.veri.be.lib.exception.ErrorCode

object ExceptionAssertions {
    fun assertApplicationException(
        executable: Executable,
        expected: ErrorCode
    ): ApplicationException {
        val exception = assertThrows(ApplicationException::class.java, executable)
        assertThat(exception.errorCode).isEqualTo(expected)
        return exception
    }
}
